ceno_zkvm/src/tables/ram.rs

```diff
@@ -80,32 +75,15 @@ impl NonVolatileTable for PubIOTable {
     const RAM_TYPE: RAMType = RAMType::Memory;
     const V_LIMBS: usize = 1; // See `MemoryExpr`.
     const WRITABLE: bool = false;
-    const OFFSET_ADDR: Addr = CENO_PLATFORM.public_io_start();
-    const END_ADDR: Addr = CENO_PLATFORM.public_io_end() + 1;
+
+    fn len() -> usize {
+        // TODO: take as program parameter.
+        1 << 2 // words
+    }
 
     fn name() -> &'static str {
         "PubIOTable"
     }
 }
 
 pub type PubIOCircuit<E> = PubIORamCircuit<E, PubIOTable>;
-
-pub fn initial_registers() -> Vec<MemInitRecord> {
-    RegTable::init_state()
-}
-
-pub fn init_program_data(program_data_content: &[Word]) -> Vec<MemInitRecord> {
-    let mut program_data_init = ProgramDataTable::init_state();
-    for (i, value) in program_data_content.iter().enumerate() {
-        program_data_init[i].value = *value;
-    }
-    program_data_init
-}
-
-pub fn init_public_io(init_public_io: &[Word]) -> Vec<MemInitRecord> {
-    let mut pubio_table = PubIOTable::init_state();
-    for (i, value) in init_public_io.iter().enumerate() {
-        pubio_table[i].value = *value;
-    }
-    pubio_table
-}
```

Comment:
This initialization functionality moved one level up into `mmu.rs` and it was generalized into ` MemPadder`.